tasks/volumes.yml

```diff
@@ -27,11 +27,25 @@
     {% endif %}
     -a {{ ansible_check_mode }}
   with_items: "{{ volumes }}"
-  when: item.type | default(libvirt_volume_default_type) == 'volume'
+  when: item.type | default(libvirt_volume_default_device) in ['volume', 'network']
   environment: "{{ libvirt_vm_script_env }}"
   register: volume_result
   changed_when:
     - volume_result is success
     - (volume_result.stdout | from_json).changed | default(True)
   check_mode: False
   become: true
+
+- name: Ensure the VM network volumes exist
+  command: qemu-img create -f {{ item.source.protocol }} {{ item.source.protocol }}:{{ item.source.name }} {{ item.capacity }}
+  with_items: "{{ volumes }}"
+  when: item.type | default(libvirt_volume_default_type) == 'network'
+  register: volume_result_network
+  # 0 is OK, 1 is an existing image
+  failed_when: volume_result_network.rc >= 2
+  changed_when:
+    - volume_result_network is success
+    - (volume_result_network.stdout | from_json).changed | default(True)
```

Comment:
Does this command output JSON? Don't we just want this:

`volume_result_network.rc == 1`

Reply:
I need to investigate this, other feedback has been incorporated.

Reply:
I've checked and qemu-img is not outputting json; I've applied your suggested change.